### PR TITLE
Fix Backend primitive classes for BackendV1 with no max_experiments

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

The ``max_experiments`` field in the BackendConfiguration for a BackendV1 backend is not a required field. While in practice most real backends set it, some simulators (including aer) do not. This causes a failure when using the Backend primitive classes with these backends as we were previously assuming the ``max_experiments`` attribute was always present.

### Details and comments

### CODE REVIEW

**Review by t-imamichi — DISMISSED** [review]
Thank you for a very rapid fix.

File: test/python/primitives/test_backend_estimator.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 from qiskit.providers.fake_provider import FakeNairobi, FakeNairobiV2
 from qiskit.quantum_info import SparsePauliOp
 from qiskit.test import QiskitTestCase
+from qiskit.utils.optionals import HAS_AER
```
Comment (t-imamichi) [inline test/python/primitives/test_backend_estimator.py:28]: Is this necessary?

Comment (mtreinish) [reply test/python/primitives/test_backend_estimator.py:28]: It is not, I originally wrote the test using AerSimulator but then switched how the test worked

```suggestion
```

File: test/python/primitives/test_backend_estimator.py [hunk 2]
```diff
@@ -297,6 +298,31 @@ def test_job_size_limit_v1(self):
             estimator.run([qc] * k, [op] * k, params_list).result()
         self.assertEqual(run_mock.call_count, 10)
 
+    def test_no_max_circuits(self):
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply test/python/primitives/test_backend_estimator.py:300]: It's not required anymore, it was a leftover from an earlier version of the test which used `AerSimulator` directly (like what I did in the neko test). But I changed how the test worked

File: qiskit/primitives/backend_estimator.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def _run_circuits(
         metadata.append(circ.metadata)
         circ.metadata = {}
     if isinstance(backend, BackendV1):
-        max_circuits = backend.configuration().max_experiments
+        max_circuits = getattr(backend.configuration(), "max_experiments", None)
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply qiskit/primitives/backend_estimator.py:59]: Yeah, I agree it's weird but I think it's probably defined behavior for the interface at this point. I did check the schema that `BackendConfiguration` was originally based on and `max_experiments` isn't a required field: https://github.com/Qiskit/ibm-quantum-schemas/blob/main/schemas/backend_configuration_schema.json#L32 which was mirrored in the class when the schema was still coupled with the class definition. It's definitely a weird interface though and why I've been so keen to get everything on `BackendV2`.

**Review by pedrorrivero — APPROVED** [review]
Thanks so much for putting this together @mtreinish !

**Review by kdk — APPROVED** [review]

**Pull request merged**